guides/javascript.md

```diff
@@ -0,0 +1,242 @@
+
+🚫
+
+``` javascript
+function isEIP1559Compatible() {
+  // ...
+}
+
+const isEIP1559Compatible = isEIP1559Compatible();
+```
+
+✅
+
+``` javascript
+function getEIP1559Compatibility() {
+  // ...
+}
+
+const isEIP1559Compatible = getEIP1559Compatibility();
+```
+
+### Read more
+
+- [Google's JavaScript Style Guide](https://google.github.io/styleguide/jsguide.html#naming-method-names)
+- [.NET Fundamentals](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods)
+- [Kotlin Coding Conventions](https://kotlinlang.org/docs/coding-conventions.html#choose-good-names)
+- [Swift API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/#strive-for-fluent-usage)
+
+## Use a verb phrase for boolean variable names that do not involve secondary objects
+
+Sometimes a variable that is intended to hold a boolean value does not have an explicit subject, but represents the context where the variable is defined (e.g., a class or an entire file). When naming such a variable, use a statement that describes the state of the subject, minus the name of the subject itself. Usually this means prefixing the name with a form of "to be" or "to have" (e.g. `is*`, `has*`), but you may find it more readable to use past or future tense and/or a modal verb such as `should`.
+
+🚫
+
+``` javascript
+const removed = false;
+```
+
+✅
+
+``` javascript
+// Any of these would do
+const isRemoved = false;
+const wasRemoved = false;
+const hasBeenRemoved = false;
+const shouldBeRemoved = false;
+```
+
+If the name represents a negative statement, reword it into a positive statement and inverting the value assigned to the value.
+
+🚫
+
+``` javascript
+const notEnoughGas = false;
+```
+
+✅
+
+``` javascript
+const hasEnoughGas = true;
+```
+
+Take special note of variables which are created via React's `useState` hook.
+
+🚫
+
+``` javascript
+const [removed, setRemoved] = useState(false);
+```
+
+✅
+
+``` javascript
+const [isRemoved, setIsRemoved] = useState(false);
+```
+
+## 💡 Place names of secondary concepts first in boolean variable names
+
+When naming a boolean variable that includes a subject, the previous guideline suggests that you can place the verb at the beginning:
+
+``` javascript
+const isRecipientOwnedAccount = Boolean(ownedAccountName);
+```
+
+However, this naming strategy creates a point of friction for objects, arrays, or React components, where it may be desirable to sort identifiers alphabetically. In that case you could end up with something like:
+
+``` jsx
+<SenderToRecipient
+  isRecipientOwnedAccount={isRecipientOwnedAccount}
+  onClick={onClick}
+  recipientName={toName}
+  recipientNickname={toNickname}
+  senderAddress={fromAddress}
+  senderName={fromName}
+/>
+```
+
+It is potentially more "greppable" if properties that concern the same concept are kept together instead of separate. To address this, you may wish to place the subject of the variable name at the beginning:
```

Comment:
Nit: This might me more readable (:sweat_smile: ) if we say this is easier to read, rather than saying "greppable". I'm not sure it really helps with code searching, given that you could do a case-insensitive search for "recipient" and still find it either way. But the readability difference is clear

```suggestion
It is potentially easier to read if properties that concern the same concept are kept together instead of separate. To address this, you may wish to place the subject of the variable name at the beginning:
```

Reply:
In using "greppable" I borrowed @brad-decker's wording from the [discussion above](https://github.com/MetaMask/contributor-docs/pull/11#discussion_r995999516). After reading it again, though, I wonder if he might have meant "scannable"/"readable" instead. So I like this suggestion, but considering that you've already approved I will make this change in a future PR (pending possible confirmation).

Reply:
I had to push up another commit anyway so I added your suggestion.